libhoney.go

```diff
@@ -120,6 +121,38 @@ type Config struct {
 	Transport http.RoundTripper
 }
 
+// VerifyWriteKey calls out to the Honeycomb API to validate the write key, so
+// we can exit immediately if desired instead of happily sending events that
+// are all rejected.
+func VerifyWriteKey(config Config) error {
+	if config.WriteKey == "" {
+		return errors.New("Write key is empty")
+	}
+	url := fmt.Sprintf("%s/1/team_slug", config.APIHost)
```

Comment:
Should probably use `defaultAPIHost` if `config.APIHost` is the empty string here.